reports/tasks.py

```diff
@@ -127,3 +127,89 @@ def send_trial_instances_report(recipients=settings.TRIAL_INSTANCES_REPORT_RECIP
     email.attach_file(csv_filename)
 
     return email.send()
+
+
+# Run on the 1st of every month
+@db_periodic_task(
+    crontab(
+        minute=TRIAL_INSTANCES_REPORT_SCHEDULE_MINUTE,
+        hour=TRIAL_INSTANCES_REPORT_SCHEDULE_HOUR + 2,
+        day=TRIAL_INSTANCES_REPORT_SCHEDULE_DAY,
+        month=TRIAL_INSTANCES_REPORT_SCHEDULE_MONTH,
+        day_of_week=TRIAL_INSTANCES_REPORT_SCHEDULE_DAY_OF_WEEK
+    )
+)
+def send_all_instances_report(recipients=settings.TRIAL_INSTANCES_REPORT_RECIPIENTS):
```

Comment:
@janimo, since this function is duplicating the task above, can you use something like the solution documented [here](https://stackoverflow.com/questions/51977211/how-to-create-a-task-function-with-an-argument-using-huey) to refactor the common code and just have a toggle that determines whether to restrict the queryset to the instances created in the last completed month? The code will become simpler with that change.

Reply:
@lgp171188 I agree that since much of the code is duplicated refactoring would make sense. Besides the queryset, the email subject and the error messages should be different.